packages/quantum-js-cli/__test__/runner.test.js

```diff
@@ -0,0 +1,228 @@
+    })
+    test("Testing evaluateInput() with input 'toDiagram' and an empty circuit", () => {
+        runner.evaluateInput("toDiagram", circuit);
+        expectedText = circuit.toDiagram();
+        expect(console.log).toHaveBeenCalled();
+        expect(console.log).toHaveBeenCalledWith(expectedText);
+    })
+    test("Testing evaluateInput() with input 'toLaTeX' and an empty circuit", () => {
+        runner.evaluateInput("toLaTeX", circuit);
+        expectedText = circuit.toLatex();
+        expect(console.log).toHaveBeenCalled();
+        expect(console.log).toHaveBeenCalledWith(expectedText);
+    })
+    test("Testing evaluateInput() with input 'report' and an empty circuit", () => {
+        runner.evaluateInput("report", circuit);
+        expectedText = circuit.report$();
+        expect(console.log).toHaveBeenCalled();
+        expect(console.log).toHaveBeenCalledWith(expectedText);
+    })
+    test("Testing evaluateInput() with input 'toText' and an empty circuit", () => {
+        runner.evaluateInput("toText", circuit);
+        expectedText = circuit.toAmazonBraket();
+        expect(console.log).toHaveBeenCalled();
+        expect(console.log).toHaveBeenCalledWith(expectedText);
+    })
+    test("Testing evaluateInput() with input 'clear' and an empty circuit", () => {
+        console.clear = jest.fn();
+        runner.evaluateInput("clear", circuit);
+        expectedText = circuit.toAmazonBraket();
+        expect(console.clear).toHaveBeenCalled();
+    })
+})
+
+//Gate operation syntax is of the regex form /(\w+\(\s*\d+\s*,\s*\[(\s*\d+\s*,{0,1}\s*)+\](,\s*\d+\.{0,1}\d+\s*)*\))/g
+//or more easily described: 
+//'gate-symbol(moment-index, [registerIndex0, registerIndex1,...], parameter0, parameter1,...)' with white space allowed liberally
+describe("Testing various forms of gate-operation call expression and see that evaluateOperation is called", ()=> {
+    //create empty circuit.
+    test("Check that evaluateOperation is called once for input 'h(1, [1])'", () => {
+        let circuit = quantumjs.Q();
+        runner.evaluateInput("h(1, [1])", circuit);
+        expect(circuit.get(1, 1).gate.symbol).toBe('H');
+    })
+    //messing with the white space
+    test("Check that evaluateOperation is called once for input 'h( 1 , [ 1 ])'", () => {
+        let circuit = quantumjs.Q();
+        runner.evaluateInput("h( 1 , [ 1 ])", circuit)
+        expect(circuit.get(1, 1).gate.symbol).toBe('H');
+    })
+    //while this doesn't update the circuit, it should still call evaluateOperation which detects the flaw later.
+    test("Check that no operation is added for 'h( 1 , [ 1 ], 3)' as the Hadamard operation takes no parameters", () => {
+        let circuit = quantumjs.Q();
+        runner.evaluateInput("h( 1 , [ 1 ], 3)", circuit)
+        expect(circuit.get(1, 1)).toBeUndefined();
+    }) 
+    test("Check that evaluateOperation is called for input 'x(1, [1, 2])'", () => {
```

Comment:
Using mocks you can assert whether a function  is called or not https://jestjs.io/docs/mock-functions Your test description talks about it is checking whether a fucntion is called or not, but it is not really checking that, use mocks to test that

Reply:
Ah, I wasn't sure how to test user prompts! Thank you, I'll look into mocking prompt()!